libs/postgres_ffi/src/lib.rs

```diff
@@ -229,7 +229,8 @@ pub use v14::bindings::{TimeLineID, TimestampTz, XLogRecPtr, XLogSegNo};
 pub use v14::bindings::{MultiXactOffset, MultiXactStatus};
 pub use v14::bindings::{PageHeaderData, XLogRecord};
 pub use v14::xlog_utils::{
-    XLOG_SIZE_OF_XLOG_LONG_PHD, XLOG_SIZE_OF_XLOG_RECORD, XLOG_SIZE_OF_XLOG_SHORT_PHD,
+    SIZEOF_CHECKPOINT, XLOG_SIZE_OF_XLOG_LONG_PHD, XLOG_SIZE_OF_XLOG_RECORD,
```

Comment:
I guess SIZEOF_CHECKPOINT happens to be the same for all supported Postgres versions, but it's a pretty shaky assumption. It's pretty shaky for all of these constants, but especially for sizeof(Checkpoint), I'm actually surprised it hasn't changed.

Could we have compile-time assertions that these constants are the same for all versions?

Reply:
Do you mean something like this in libs/postgres_ffi/src/lib.rs:
```
static_assert!(v14::xlog_utils::SIZEOF_CHECKPOINT == v15::xlog_utils::SIZEOF_CHECKPOINT && v15::xlog_utils::SIZEOF_CHECKPOINT == v16::xlog_utils::SIZEOF_CHECKPOINT && v16::xlog_utils::SIZEOF_CHECKPOINT == v17::xlog_utils::SIZEOF_CHECKPOINT);
```

But as afar as I undertstand `static_assert` is not yet available in Rust:
https://github.com/rust-lang/libs-team/issues/325

Reply:
Yeah, something like that. There's some precedence for this in `pageserver/src/assert_u64_eq_usize.rs`

Reply:
Should be part of separate PR